pkg/elb/cleanup.go

```diff
@@ -0,0 +1,299 @@
+package elb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
+	"github.com/kris-nova/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	awsprovider "k8s.io/kubernetes/pkg/cloudprovider/providers/aws"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/eks"
+)
+
+const (
+	loadBalancerKindClassic = iota
+	loadBalancerKindNetwork = iota
+)
+
+type loadBalancer struct {
+	kind                  int
+	ownedSecurityGroupIDs map[string]struct{}
+}
+
+// Cleanup finds and deletes any dangling ELBs associated to a Kubernetes Service
+func Cleanup(ctx context.Context, ec2API ec2iface.EC2API, elbAPI elbiface.ELBAPI, elbv2API elbv2iface.ELBV2API,
+	client *eks.Client, clusterConfig *api.ClusterConfig) error {
+
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("not context deadline set in call to elb.Cleanup()")
+	}
+
+	kubernetesCS, err := client.NewClientSet()
+	if err != nil {
+		return err
+	}
+	services, err := kubernetesCS.CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Delete Services of type 'LoadBalancer', collecting their ELBs to wait for them to be deleted later on
+	loadBalancers := map[string]loadBalancer{}
+	for _, s := range services.Items {
+		if s.Spec.Type == corev1.ServiceTypeLoadBalancer {
+			name := cloudprovider.DefaultLoadBalancerName(&s)
```

Comment:
Maybe splitting lines 59-79 into their own function we can make testing easier

Reply:
I will address that in #1031